game/src/test/scala/io/aigar/game/ResourcesSpec.scala

```diff
@@ -14,30 +14,43 @@ class ResourcesSpec extends FlatSpec with Matchers {
   }
 
   it should "respawn resources when the quantity is minimal" in {
-    val resources = new Resources(new Grid(0, 0))
+    val game = new Game(0, 1 to 15 toList)
+    val resources = new Resources(game.grid)
 
     for(resourceType <- resources.resourceTypes){
       resourceType.positions = resourceType.positions.take(resourceType.min)
     }
 
-    resources.update
+    resources.update(game.players)
 
     for(resourceType <- resources.resourceTypes){
       resourceType.positions.length should be > resourceType.min
     }
   }
 
   it should "not respawn resources when the quantity is maximal" in {
-    val resources = new Resources(new Grid(0, 0))
+    val game = new Game(0, 1 to 15 toList)
+    val resources = new Resources(game.grid)
 
     for(resourceType <- resources.resourceTypes){
       resourceType.positions = List.fill(resourceType.max)(new Grid(0, 0).randomPosition)
     }
 
-    resources.update
+    resources.update(game.players)
 
     for(resourceType <- resources.resourceTypes){
       resourceType.positions.length should equal(resourceType.max)
     }
   }
+
+  "Cell" should "increase its mass when eating regular or silver resources" in {
+    val game = new Game(0, 1 to 15 toList)
+    val resources = new Resources(game.grid)
+    val cell = game.players.head.cells.head
+    val initialMass = cell.mass
+
+    resources.resourceTypes.head.reward(cell, 1, 0)
```

Comment:
Very smarter way to test! I've implemented both tests. Please, review them and tell me what you think about it!